### PR TITLE
refactor(test): move image utils for tests in a separate module

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/test/common.go [hunk 1]
```diff
@@ -239,7 +235,7 @@ func CopyTestKeysAndCerts(destDir string) error {
 		"client.key", "server.cert", "server.csr", "server.key",
 	}
 
-	rootPath, err := GetProjectRootDir()
+	rootPath, err := image.GetProjectRootDir()
```
Comment (andaaron) [inline pkg/test/common.go:238]: This starts being confusing.
We're either using image, imageUtils, or dot imports.
It's too many variants for the same import. how about we use imageUtils or dot imports? Just 1 of these 2 variants.

File: pkg/test/common.go [hunk 2]
```diff
@@ -327,7 +323,7 @@ func NewControllerManager(controller Controller) ControllerManager {
 	return cm
 }
 
-func WriteImageToFileSystem(image Image, repoName, ref string, storeController storage.StoreController) error {
+func WriteImageToFileSystem(image image.Image, repoName, ref string, storeController storage.StoreController) error {
```
Comment (andaaron) [inline pkg/test/common.go:326]: Why not move this to the image-utils folder?

File: pkg/test/common.go [hunk 3]
```diff
@@ -371,7 +367,7 @@ func WriteImageToFileSystem(image Image, repoName, ref string, storeController s
 	return nil
 }
 
-func WriteMultiArchImageToFileSystem(multiarchImage MultiarchImage, repoName, ref string,
+func WriteMultiArchImageToFileSystem(multiarchImage image.MultiarchImage, repoName, ref string,
```
Comment (andaaron) [inline pkg/test/common.go:370]: Can we move this to the new package?

File: pkg/test/common.go [hunk 4]
```diff
@@ -873,27 +788,27 @@ func GetCosignSignatureTagForDigest(manifestDigest godigest.Digest) string {
...
-	return Image{
+	return image.Image{
 		Manifest: manifest,
 		Config:   conf,
 		Layers:   layers,
 	}, nil
 }
 
-func UploadImage(img Image, baseURL, repo, ref string) error {
+func UploadImage(img image.Image, baseURL, repo, ref string) error {
```
Comment (andaaron) [inline pkg/test/common.go:811]: Can we move this to the new package?

File: pkg/test/common.go
Comment (andaaron) [inline pkg/test/common.go:1]: Let's review all functions in this file and decide what to do about them.

File: pkg/test/common_test.go [hunk 5]
```diff
@@ -932,7 +436,7 @@ func TestInjectUploadImageWithBasicAuth(t *testing.T) {
 			}
 		}
 
-		img := test.Image{
```
Comment (andaaron) [inline pkg/test/common_test.go:935]: TestInjectUploadImageWithBasicAuth should also be under the upload tests in the new folder, right?

File: pkg/test/common.go [hunk 6]
```diff
@@ -632,18 +583,11 @@ func GetRandomImageComponents(layerSize int) (ispec.Image, [][]byte, ispec.Manif
 	return config, layers, manifest, nil
 }
 
-// These are the 3 vulnerabilities found for the returned image by the GetVulnImage function.
-const (
```
Comment (andaaron) [inline pkg/test/common.go:636]: Maybe move these to image.go (to be together with the creation of the vulnerable images) instead of upload.go.

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**